url75694980d5_fabioalmeid_tp01ftc_tgz-pJ8-agentes_jade_Centralizador_ExemploParserGramaticaCentralizadorJ8/src/monitor/PrettyPrinter.java

```diff
@@ -0,0 +1,408 @@
+  {
+    sh(foo);
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  public static String print(monitor.Absyn.Acao foo)
+  {
+    pp(foo, 0);
+    trim();
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  public static String show(monitor.Absyn.Acao foo)
+  {
+    sh(foo);
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  public static String print(monitor.Absyn.Dados foo)
+  {
+    pp(foo, 0);
+    trim();
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  public static String show(monitor.Absyn.Dados foo)
+  {
+    sh(foo);
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  public static String print(monitor.Absyn.Quantidade foo)
+  {
+    pp(foo, 0);
+    trim();
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  public static String show(monitor.Absyn.Quantidade foo)
+  {
+    sh(foo);
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  public static String print(monitor.Absyn.Operador foo)
+  {
+    pp(foo, 0);
+    trim();
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  public static String show(monitor.Absyn.Operador foo)
+  {
+    sh(foo);
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  public static String print(monitor.Absyn.Hora foo)
+  {
+    pp(foo, 0);
+    trim();
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  public static String show(monitor.Absyn.Hora foo)
+  {
+    sh(foo);
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  /***   You shouldn't need to change anything beyond this point.   ***/
+
+  private static void pp(monitor.Absyn.Tarefa foo, int _i_)
+  {
+    if (foo instanceof monitor.Absyn.ETarefa)
+    {
+       monitor.Absyn.ETarefa _etarefa = (monitor.Absyn.ETarefa) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       pp(_etarefa.acao_, 0);
+       if (_i_ > 0) render(_R_PAREN);
+    }
+  }
+
+  private static void pp(monitor.Absyn.Acao foo, int _i_)
+  {
+    if (foo instanceof monitor.Absyn.EAcao)
+    {
+       monitor.Absyn.EAcao _eacao = (monitor.Absyn.EAcao) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       pp(_eacao.dados_, 0);
+       render("as");
+       pp(_eacao.hora_, 0);
+       if (_i_ > 0) render(_R_PAREN);
+    }
+    else     if (foo instanceof monitor.Absyn.EAcao2)
+    {
+       monitor.Absyn.EAcao2 _eacao2 = (monitor.Absyn.EAcao2) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       pp(_eacao2.acao_1, 0);
+       pp(_eacao2.operador_, 0);
+       pp(_eacao2.acao_2, 0);
+       if (_i_ > 0) render(_R_PAREN);
+    }
+  }
+
+  private static void pp(monitor.Absyn.Dados foo, int _i_)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.